packages/netlify/src/index.ts

```diff
@@ -24,6 +24,90 @@ const isStaticRedirect = (route: RouteData) =>
 
 const clearDirectory = (dir: URL) => rm(dir, { recursive: true }).catch(() => {});
 
+type RemotePattern = AstroConfig['image']['remotePatterns'][number];
+
+/**
+ * Convert a remote pattern object to a regex string
+ */
+export function remotePatternToRegex({
+	protocol,
+	hostname,
+	port,
+	pathname,
+}: RemotePattern): string {
+	let regexStr = '';
+
+	if (protocol) {
+		regexStr += `${protocol}://`;
+	} else {
+		// Default to matching any protocol
+		regexStr += '[a-z]+://';
+	}
+
+	if (hostname) {
+		if (hostname.startsWith('**.')) {
+			// match any number of subdomains
+			regexStr += '([a-z0-9]+\\.)*';
+			hostname = hostname.substring(3);
+		} else if (hostname.startsWith('*.')) {
+			// match one subdomain
+			regexStr += '([a-z0-9]+\\.)?';
+			hostname = hostname.substring(2); // Remove '*.' from the beginning
+		}
+		// Escape dots in the hostname
+		regexStr += hostname.replace(/\./g, '\\.');
+	} else {
+		regexStr += '[a-z0-9.-]+';
+	}
+
+	if (port) {
+		regexStr += `:${port}`;
+	} else {
+		// Default to matching any port
+		regexStr += '(:[0-9]+)?';
+	}
+
+	if (pathname) {
+		if (pathname.endsWith('/**')) {
+			// Match any path. Return early because there's no need to add more to the regex
+			return `${regexStr}(\\${pathname.replace('/**', '')}.*)`;
+		}
+		if (pathname.endsWith('/*')) {
+			// Match one level of path
+			regexStr += `(\\${pathname.replace('/*', '')}\/[^/?#]+)\/?`;
+		} else {
+			// Exact match
+			regexStr += `(\\${pathname})`;
+		}
+	} else {
+		// Default to matching any path
+		regexStr += '(\\/[^?#]*)?';
+	}
+	// Match query
+	regexStr += '([?][^#]*)?';
+	return regexStr;
+}
+
+async function writeNetlifyDeployConfig(config: AstroConfig) {
+	const remoteImages: Array<string> = [];
+	// Domains get a simple regex match
+	remoteImages.push(
+		...config.image.domains.map((domain) => `https?:/\/${domain.replaceAll('.', '\\.')}\/.*`)
+	);
+	// Remote patterns need to be converted to regex
+	remoteImages.push(...config.image.remotePatterns.map(remotePatternToRegex));
```

Comment:
Isn't there a way to check if the regex isn't malformed before using it? After all it's built dynamically with a lot of branching

Reply:
That's a good idea

Reply:
I guess the question is what we do with that information if it is malformed. Throw? Log a warning?

Reply:
I suppose we can log an error/warning to the user, and maybe logging the produced regex will help to identify a possible bug. I don't think it's worth blocking the build, right?

Reply:
@ematipico I've added a warning log. See if it looks right to you